src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -55,6 +58,31 @@ export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ chil
 
       const { getOpportunitiesUserData } = opportunitiesApi.endpoints
 
+      const idleContractAddresses = [
+        ...Object.values(stakingOpportunitiesById)
+          .map(opportunity => (opportunity as IdleStakingSpecificMetadata | undefined)?.cdoAddress)
+          .filter(isSome),
```

Comment:
I'd prefer to see this pulled out as a separate constant for quicker comprehension.

Reply:
05d16675cbf1c31cb156a0d97cba30b3d6b05032